SIPS/sip-63.md

```diff
@@ -28,21 +28,80 @@ There is currently no way to track trading fees paid on-chain, so it is not poss
+We considered a number of approaches to this problem, including minting fee tokens and paying incentives immediately on each trade. There are several benefits to this approach over the alternatives considered. The first is that users can accumulate as many fees as they want before claiming reducing gas costs and allowing even small traders to wait until they have sufficient fees to justify a withdrawal. Secondly this negates the requirement of handling fee period rollovers or unclaimed fees as trading incentives are not dependent on feepool periods. Each week the balanance of fees paid is wiped when the new SNX deposit is made ensuring all traders start each fee period with an equal chance to earn fees.
 
 ### Technical Specification
 <!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Synthetix currently exposes or the creations of new ones.-->
 The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Synthetix currently exposes or the creations of new ones. 
 
+**Trading Incentive Contract**
+
+When an exchange is made by a user on Synthetix Exchange, the amount of fees remitted (in sUSD) and user the funds are exchanged `from` will be recorded in the `Trading incentive contract`. Each `address` will accumulate the total of exchange fees that they generate for a period (tracked as a `periodID`) and the counter is reset when SNX for the incentives is deposited for the period and a user makes another trade. The total of the previous period will be stored on the next trade regardless of how long ago the last trade was.   
+
+For exchanges occuring through integrations and exchange aggregators such as `1inch` there will be a function that allows an `address` to be passed through the exchange to be recorded against the originating address for the trading volume incentives. 
+
+The `trading incentive` contract will track the `totalFeesInPeriod` for each period generated by exchanges, which will be used for calculating the % of the fees an account has generated in that period and hence the % of the SNX rewards they will be allocated. 
+
+The snx rewards an account can claim for a period is calculated as:
+
+\\[ reward = \frac{Account's Exchange Fees for period}{Total Exchange Fees for period} * SNX rewards for period 
+\\]
+
+The `totalFeesInPeriod` will be stored and kept for each period once the new SNX is deposited for that period and closes it, resetting the balances for the next period to accumulate. 
+
+This allows the account's to claim their trading incentives in any period they want in the future and combining multiple periods to claim.
+
+**Interface**
+
+```
+pragma solidity >=0.4.24;
+
+interface ITradingIncentive {
+  // Views
+  function rewards(uint periodID) external view; 
+  
+  function rewardForPeriods(uint[] periodIDs) external view;
```

Comment:
Wouldn't you also have a view so an account can see their specific rewards?

Reply:
Good idea, there should be account specific views

Reply:
updated